src/context/client.rs

```diff
@@ -19,48 +20,92 @@ impl RedisModuleClientInfo {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById using current client ID
+    pub fn get_client_name(&self) -> ValkeyString {
+        self.get_client_name_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_SetClientNameById
+    pub fn set_client_name_by_id(&self, client_id: u64, client_name: &ValkeyString) -> i64 {
         let resp = unsafe { RedisModule_SetClientNameById.unwrap()(client_id, client_name.inner) };
         resp as i64
     }
 
-    pub fn get_client_username(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
-        let client_username =
-            unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
-        ValkeyString::from_redis_module_string(self.ctx, client_username)
+    /// wrapper for RedisModule_SetClientNameById using current client ID
+    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
+        self.set_client_name_by_id(self.get_client_id(), client_name)
     }
 
+    /// wrapper for RedisModule_GetClientUserNameById
     pub fn get_client_username_by_id(&self, client_id: u64) -> ValkeyString {
         let client_username =
             unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_username)
     }
 
+    /// wrapper for RedisModule_GetClientUserNameById using current client ID
+    pub fn get_client_username(&self) -> ValkeyString {
+        self.get_client_username_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_GetClientCertificate
     pub fn get_client_cert(&self) -> ValkeyString {
         let client_id = self.get_client_id();
         let client_cert = unsafe { RedisModule_GetClientCertificate.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_cert)
     }
 
-    pub fn get_client_info(&self) -> RedisModuleClientInfo {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientInfoById
+    pub fn get_client_info_by_id(&self, client_id: u64) -> RedisModuleClientInfo {
         let mut mci = RedisModuleClientInfo::new();
         let mci_ptr: *mut c_void = &mut mci as *mut _ as *mut c_void;
         unsafe { RedisModule_GetClientInfoById.unwrap()(mci_ptr, client_id) };
         mci
     }
+
+    /// wrapper for RedisModule_GetClientInfoById using current client ID
+    pub fn get_client_info(&self) -> RedisModuleClientInfo {
+        self.get_client_info_by_id(self.get_client_id())
+    }
+
+    /// wrapper to get the client IP address from RedisModuleClientInfo
+    pub fn get_client_ip_by_id(&self, client_id: u64) -> String {
+        let client_info = self.get_client_info_by_id(client_id);
+        let addr_u8: Vec<u8> = client_info.addr.iter().map(|&x| x as u8).collect();
+        let ip_addr_as_string = String::from_utf8_lossy(&addr_u8)
+            .trim_matches(char::from(0))
```

Comment:
Question: Why is this needed?

Reply:
I am planning to use it for the IP based auth solution that I demoed last week.  I thought it could be useful to others so I added in the SDK

Reply:
Yes, the API itself makes sense.

But I was asking about `.trim_matches(char::from(0))` in specific. Do we need it to remote a trailing 0 ?

Reply:
without trim there would be a bunch of 0s at the end of the string